adafruit_ble_eddystone/__init__.py

```diff
@@ -116,17 +116,10 @@ def __init__(self, *, minimum_size=None):
         self.connectable = False
         self.flags.general_discovery = True
         self.flags.le_only = True
-        self.frame_type = bytearray(1)
-        # Frame type is in the prefix.
-        self.frame_type[0] = self.prefix[-1]
+        # self.frame_type is defined by the subclass.
         if not self.eddystone_frame:
             self.eddystone_frame = bytearray(minimum_size)
 
-    @classmethod
-    def matches(cls, entry):
-        """True if the entry matches all of the prefixes. This is stricter than the scan."""
```

Comment:
Why can you remove this?

Reply:
Unless I'm misunderstanding, this is redundant, because
https://github.com/adafruit/Adafruit_CircuitPython_BLE/blob/5d584576ef79ca36506e6c7470e7ac5204cf0a8d/adafruit_ble/advertising/__init__.py#L274-L281:
```python
    @classmethod
    def matches(cls, entry):
        """Returns true if the given `_bleio.ScanEntry` matches all portions of the Advertisement
           type's prefix."""
        if not hasattr(cls, "prefix"):
            return True

        return entry.matches(cls.prefix)
```
uses the default value of `entry.matches()`, which is `(..., all=True)`:
https://github.com/adafruit/circuitpython/blob/9285252fbcbe4db654bc69621210883380fb6981/shared-bindings/_bleio/ScanEntry.c#L48-L60
```c
//|     def matches(self, prefixes: Any, *, all: Any = True) -> Any:
//|         """Returns True if the ScanEntry matches all prefixes when ``all`` is True. This is stricter
//|         than the scan filtering which accepts any advertisements that match any of the prefixes
//|         where all is False."""
//|         ...
//|
STATIC mp_obj_t bleio_scanentry_matches(mp_uint_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
    bleio_scanentry_obj_t *self = MP_OBJ_TO_PTR(pos_args[0]);

    enum { ARG_prefixes, ARG_all };
    static const mp_arg_t allowed_args[] = {
        { MP_QSTR_prefixes, MP_ARG_OBJ | MP_ARG_REQUIRED },
        { MP_QSTR_all, MP_ARG_BOOL | MP_ARG_KW_ONLY, {.u_bool = true} },
```